disaster-recovery/src/test/scala/uk/gov/nationalarchives/disasterrecovery/MainTest.scala

```diff
@@ -380,7 +368,7 @@ class MainTest extends AnyFlatSpec with MockitoSugar with EitherValues {
       repo.getObject(ioId.toHeadVersion).containsFile(path) must be(true)
     }
 
-    utils.latestObjectVersion(repo, ioId) must equal(2)
+    utils.latestObjectVersion(repo, ioId) must equal(4)
```

Comment:
Is the reason why this increased from 2 to 4 because since each call to `createObjects` is done by a different fiber, objects that came from different messages (in the batch) are no longer getting the same version? If that's the case, the this new version is actually more accurate than it was before as now, if, in a batch, there are multiple COs (belonging to the same IO) to be added/updated, it will create a new version for each CO added/updated, which is what we want, right?

Reply:
It's less the fibers and more that we're not grouping the messages and running them together any more so you're right, we get one version per file. I don't think we mind really if we get multiple files in a version. This  is a side effect of running `process` once per message so we don't get messages that would have been succesful failing because there in a list with a failing message.

Reply:
In short, yes, it's fine :smile: